tasks/bal_sorswap.ts

```diff
@@ -123,49 +127,68 @@ task('bal_sorswap', 'Swap 2 tokens via Balancer SOR', async (args, hre: HardhatR
 
   const networkERC20s = getNetworkERC20s(hre);
 
+  // If true(swaps erc20 for erc20) else (swaps eth for erc20)
+  const isErc20Swap = true;
```

Comment:
Can we generalize this `bal_sorswap` task?

I hardcoded things originally just to get things working, but can we have the task take parameters like it shows here: https://hardhat.org/guides/create-task.html

```
task('bal_sorswap', 'Swap 2 tokens via Balancer SOR'
  .addParam("swap_type", "'in' or 'out'"
  .addParam("token_in", "Token in")
  .addParam("token_out", "Token out")
  .addParam("token_amount", "Token amount")
  .setAction(async () => {
      // insert task code here
  });
```

Maybe just support ETH and 3 ERC20s like BAL, USDC, AAVE or  something, otherwise throw an error message. That will help a lot when we get to more tests and test deployments.